pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -240,109 +241,220 @@ func substitueYamlVariables(baseYaml []byte, yamlSubstitutions map[string]string
+	writtenToManifest := false
+	manifestFile, err := os.Create(filepath.Join(a.Context, ".odo", "manifest.yaml"))
 	if err != nil {
-		return err
+		err = manifestFile.Close()
+		return errors.Wrap(err, "Unable to create the local manifest file")
 	}
 
-	klog.V(3).Infof("Deploy manifest:\n\n%s", deploymentManifest)
-	gvr := schema.GroupVersionResource{Group: gvk.Group, Version: gvk.Version, Resource: strings.ToLower(gvk.Kind + "s")}
-	klog.V(3).Infof("Manifest type: %s", gvr.String())
+	manifests := bytes.Split(parameters.ManifestSource, []byte("---"))
+	for _, manifest := range manifests {
+		if len(manifest) > 0 {
+			// Substitute the values in the manifest file
+			deployYaml := substitueYamlVariables(manifest, yamlSubstitutions)
 
-	labels := map[string]string{
-		"component": applicationName,
-	}
+			_, gvk, err := yamlDecoder.Decode([]byte(deployYaml), nil, deploymentManifest)
+			if err != nil {
+				return errors.Wrap(err, "Failed to decode the manifest yaml")
+			}
 
-	manifestLabels := deploymentManifest.GetLabels()
-	if manifestLabels != nil {
-		for key, value := range labels {
-			manifestLabels[key] = value
-		}
-		deploymentManifest.SetLabels(manifestLabels)
-	} else {
-		deploymentManifest.SetLabels(labels)
-	}
+			deployJSON, err := deploymentManifest.MarshalJSON()
+			if err != nil {
+				return errors.Wrap(err, "Deployment manifest is invalid")
+			}
+			klog.V(3).Infof("Deploy manifest:\n\n%s\n", deployJSON)
 
-	// TODO: Determine why using a.Client.DynamicClient doesnt work
-	// Need to create my own client in order to get the dynamic parts working
-	myclient, err := dynamic.NewForConfig(a.Client.KubeClientConfig)
-	if err != nil {
-		return err
-	}
+			gvr := schema.GroupVersionResource{Group: gvk.Group, Version: gvk.Version, Resource: strings.ToLower(gvk.Kind + "s")}
+			klog.V(3).Infof("Manifest type: %s", gvr.String())
+
+			labels := map[string]string{
+				"component": applicationName,
+			}
 
-	// Check to see whether deployed resource already exists. If not, create else update
-	// Get?
-	instanceFound := false
-	list, err := myclient.Resource(gvr).Namespace(namespace).List(metav1.ListOptions{})
-	if list != nil && len(list.Items) > 0 {
-		for _, item := range list.Items {
-			klog.V(3).Infof("Found %s %s with resourceVersion: %s.\n", gvk.Kind, item.GetName(), item.GetResourceVersion())
-			if item.GetName() == applicationName {
-				deploymentManifest.SetResourceVersion(item.GetResourceVersion())
-				instanceFound = true
+			manifestLabels := deploymentManifest.GetLabels()
+			if manifestLabels != nil {
+				for key, value := range labels {
+					manifestLabels[key] = value
+				}
+				deploymentManifest.SetLabels(manifestLabels)
+			} else {
+				deploymentManifest.SetLabels(labels)
+			}
+
+			// Check to see whether deployed resource already exists. If not, create else update
+			instanceFound := false
+			list, err := a.Client.DynamicClient.Resource(gvr).Namespace(namespace).List(metav1.ListOptions{})
```

Comment:
could we do a `Get()` instead of `List()`?

Reply:
Yes, probably. Will do that in a new PR though!

Reply:
Have made change to use Get rather than List.